homeassistant/components/fmi/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for FMI (Finnish Meteorological Institute) integration."""
+
+import fmi_weather_client as fmi_client
+from fmi_weather_client.errors import ClientError, ServerError
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME, CONF_OFFSET
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import base_unique_id
+from .const import _LOGGER, FORECAST_OFFSET
+
+
+async def validate_user_config(hass: core.HomeAssistant, data):
```

Comment:
I think the naming of this method is incorrect. It does not just validate data, it also provides data.